### PR TITLE
Add support for CRUD operations for Categories and Tags

### PR DESCRIPTION
**Pull request by lucatume**
Ticket:  https://central.tri.be/issues/82414

Based on: https://github.com/moderntribe/tribe-common/pull/452

This PR adds the code and the tests to fully support CRUD operations for event Categories and Tags.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $return_id is never used | [Term_Single_Base.php:94](https://github.com/moderntribe/the-events-calendar/blob/50e724fae2b6714cb2273e1f8dca74d14031196c/src/Tribe/REST/V1/Endpoints/Term_Single_Base.php#L94)

> via jenkins/codesniffer-phpcs

File: tests/_support/Helper/TribeDb.php [hunk 1]
```diff
@@ -225,11 +225,11 @@ public function haveEventInDatabase( array $overrides = [] ) {
 
 		$tax_input = [];
 		if ( isset( $overrides['categories'] ) ) {
-			$tax_input['tribe_events_cat'] = $overrides['categories'];
+			$tax_input['tribe_events_cat'] = (array)$overrides['categories'];
```
Comment (bordoni) [inline tests/_support/Helper/TribeDb.php]: Space after `(array)` .

File: tests/_support/Helper/TribeDb.php [hunk 2]
```diff
@@ -225,11 +225,11 @@ public function haveEventInDatabase( array $overrides = [] ) {
...
 			unset( $overrides['categories'] );
 		}
 		if ( isset( $overrides['tags'] ) ) {
-			$tax_input['post_tag'] = $overrides['tags'];
+			$tax_input['post_tag'] = (array)$overrides['tags'];
```
Comment (bordoni) [inline tests/_support/Helper/TribeDb.php]: Space after `(array)` .

File: tests/_support/Helper/TribeDb.php [hunk 3]
```diff
@@ -254,7 +254,7 @@ public function haveEventInDatabase( array $overrides = [] ) {
 
 		unset( $overrides['when'], $overrides['duration'], $overrides['utc_offset'] );
 
-		$id = uniqid();
+		$id = uniqid( mt_rand( 1, 999 ), true );
```
Comment (bordoni) [inline tests/_support/Helper/TribeDb.php]: Good stuff here! We've learned the lesson from Windows. 💘

File: src/Tribe/REST/V1/Endpoints/Single_Tag.php [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	 * Returns the content of the `args` array that should be used to register the endpoint
+	 * with the `register_rest_route` function.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function CREATE_args() {
+		return array(
+			'name'             => array(
```
Comment (bordoni) [inline src/Tribe/REST/V1/Endpoints/Single_Tag.php]: Align these `=>` back with the key, it looks weird when there are too many diff indentations.

File: src/Tribe/REST/V1/Endpoints/Single_Tag.php [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+				'required'          => true,
+				'validate_callback' => array( $this->validator, 'is_string' ),
+				'type'              => 'string',
+				'description'       => __( 'The event tag name', 'the-events-calendar' ),
+			),
+			'description'             => array(
```
Comment (bordoni) [inline src/Tribe/REST/V1/Endpoints/Single_Tag.php]: Align these `=>` back with the key, it looks weird when there are too many diff indentations.

**Comment by lucatume** [comment]
Damn arrows are in place.

**Pull request merged**